firedrake/functionspace.py

```diff
@@ -284,17 +288,25 @@ def exterior_facet_node_map(self, bcs=None):
                                parent=parent,
                                offset=offset)
 
-    def bottom_nodes(self):
+    def bottom_nodes(self, method='topological'):
         """Return a list of the bottom boundary nodes of the extruded mesh.
         The bottom mask is applied to every bottom layer cell to get the
         dof ids."""
-        return np.unique(self.cell_node_list[:, self.bt_masks[0]])
+        try:
```

Comment:
I don't think this is true in this case. If there is no [0] then you'll get an IndexError so you won't fall through to the exception handler.